src/js/modules/infragistics.ui.editors.js

```diff
@@ -9178,10 +9174,15 @@
 				}
 				return;
 			}
-			if ($.type(value) === "date") {
-				parsedVal = value;
-			} else {
-				parsedVal = this._parseDateFromMaskedValue(value);
+			parsedVal = value;

```

Comment:
there is no need of parsedVal at this point. We need to move the definition a few lines down in the if block